veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,590 @@
+
+## Functional Testing Approach
+
+The goal of testing for multi-hypervisor support is to ensure that for each in-tree hypervisor implementation, all supported KubeVirt features are verified.
+
+- Unit tests for each hypervisor implementation for validating functionality concerning each extension point with comprehensive coverage for all new code.
+- Integration tests should test the following for each in-tree hypervisor:
+  - Validate if `virt-handler`'s device plugin can detect correct hypervisor device on the node and label node with correct allocatable and capacity.
+  - Validate `virt-controller`'s rendering of `virt-launcher` pod spec.
+  - Validate node-labeller's ability to add expected node labels for each hypervisor.
+  - Validate runtime adjustments to VMI, such as memlock limit adjustment and housekeeping thread management.
+
+### Integration with existing Prow-based CI testing
+
+To ensure robust validation of the proposed in-tree Microsoft Hypervisor (MSHV) integration, we recommend incorporating MSHV-specific tests into KubeVirt’s existing CI workflows. The following changes are proposed:
+
+- Introduce dedicated testing lanes for MSHV on AMD64, aligned with each SIG (e.g., sig-compute) and Kubernetes version.
+- Enhance the Prow provisioner to support provisioning Azure-based Kubernetes clusters, enabling deployment and testing of KubeVirt distributions backed by MSHV.
+- Optimize CI resource usage by scheduling non-KVM hypervisor tests during the second phase of CI execution—triggered after the /lgtm label is applied—when comprehensive validation runs are performed.
+
+## Implementation Phases
+
+1. Refactor KubeVirt to introduce the above interfaces for the Hypervisor extension points. Implement the interface for KVM only, such that KubeVirt continues to be able to create and manage KubeVirt VMs.
+
+2. Add the `HypervisorConfiguration` CR to KubeVirt API and the `hypervisor` field to the `KubevirtConfiguration` CR. Add the `ConfigurableHypervisor` feature gate.
+
+3. For each interface, add the MSHV implementation that is gated by the `ConfigurableHypervisor` feature gate.
+
+4. Add testing scripts to enable testing KubeVirt on an MSHV platform.
+
+
+## Graduation Requirements
+
+### Alpha
+
+- Feature gate covers configurable hypervisor
+- Validation webhook for KubeVirt CR enforce that hypervisor configuration contains at most 1 entry, thereby enforcing only 1 supported hypervisor in the cluster.
+- Cluster-wide hypervisor configuration implemented and consumed by defaults, converter, and webhooks.
+- Basic functional tests for alternative hypervisor scheduling and domain generation.
+- Update of the `HypervisorConfiguration` value in KubeVirt CR after a KubeVirt deployment is running will not result in any change to the deployment. This setting is only to be used at the deployment time.
+
+### Beta
```

Comment:
What features are expected to be covered in this phase?

What about the out-of-tree move?